src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/ApiSetType.java

```diff
@@ -27,25 +26,30 @@ private ApiSetType(
     super(ApiTypeName.SET, valueType, DataTypes.setOf(valueType.cqlType(), isFrozen), apiSupport);
   }
 
+  @Override
+  public boolean isFrozen() {
+    return cqlType.isFrozen();
+  }
+
   @Override
   public ColumnDesc columnDesc() {
     return new SetColumnDesc(valueType.columnDesc(), ApiSupportDesc.from(this));
   }
 
+  /**
+   * Creates a new {@link ApiSetType} from the given ApiDataType value.
+   *
+   * <p>ApiSupport for valueType should be already validated.
+   */
   static ApiSetType from(ApiDataType valueType, boolean isFrozen) {
     Objects.requireNonNull(valueType, "valueType must not be null");
-
-    if (isValueTypeSupported(valueType)) {
-      return new ApiSetType(
-          (PrimitiveApiDataTypeDef) valueType, defaultApiSupport(isFrozen), isFrozen);
-    }
-    throw new IllegalArgumentException(
-        "valueType must be primitive type, valueType%s".formatted(valueType));
+    return new ApiSetType(
```

Comment:
-1 for a non private function assuming validation has been done, please put the isValueTypeSupported() check back